vcenter/vcenter.py

```diff
@@ -449,6 +441,88 @@ def stop(self, uuid):
+
+        self.__logger.debug(f'found vm: {vm.config.uuid}')
+        screenshot_task = vm.CreateScreenshot_Task()
+        self.wait_for_task(screenshot_task)
+        result_path = screenshot_task.info.result
+        if result_path is None:
+            raise RuntimeError('Taking screenshot failed!')
+
+        # can't we just use self.destination_datastore.info.name ?
+        datastore_name, screenshot_path = result_path.split(' ')
+        datastore_name = datastore_name.lstrip('[').rstrip(']')
+        self.__logger.debug(f'<- take_screenshot: {datastore_name}, {screenshot_path}')
+        return datastore_name, screenshot_path
+
+    # TODO rewrite others to use this one
+    def __get_objects_list_from_container(self, container, object_type):
+
+        result = []
+        object_view = None
+        try:
+            object_view = self.content.viewManager.CreateContainerView(
+                    container,
+                    [object_type],
+                    True)
+            result = list(object_view.view)
+        except vmodl.fault.ManagedObjectNotFound:
+            self.__logger.warn('vmodl.fault.ManagedObjectNotFound has occured')
+        except Exception:
+            settings.raven.captureException(exc_info=True)
+        finally:
+            if object_view is not None:
+                object_view.Destroy()
+
+        return result
+
+    def __get_datacenter_for_datastore(self, datastore_name):
+        datacenters = self.__get_objects_list_from_container(self.content.rootFolder, vim.Datacenter)
+        for dc in datacenters:
+            datastores = self.__get_objects_list_from_container(dc, vim.Datastore)
+            for ds in datastores:
+                if ds.info.name == datastore_name:
+                    return dc
+        return None
+
+    def download_file_from_datastore(self, remote_path_to_file, datastore_name, custom_local_path=None):
+        """
+        Downloads file from datastore
+        :param remote_path_to_file: path to file in datastore (e.g. my_vm/my_vm.png)
+        :param datastore_name: name of datastore
+        :param custom_local_path: Optional, is specified, file will be saved with this path. Otherwise to current working directory using the same filename as on remote
+        :return: path to downloaded file
+        """
+        server_name =  settings.app['vsphere']['host']
+        datacenter = self.__get_datacenter_for_datastore(datastore_name)
+        if datacenter is None:
+            raise RuntimeError(f'Cannot find datacenter for datastore {datastore_name}')
+
+        download_url= f'https://{server_name}/folder/{remote_path_to_file}?dcPath={datacenter.name}&dsName={datastore_name}'
+        resp = requests.get(url=download_url, verify=False, headers={'Cookie': self._connection_cookie})
+
+        if resp.status_code != 200:
+            raise RuntimeError(f'Cannot download file: {remote_path_to_file} (from datastore: {datastore_name})!')
```

Comment:
is it ok to throw exceptions like this in general here? Or do I need to handle them and just log error and reflect error state in method return value?